### PR TITLE
fix commonjs js provide GOOGLE_CHECK

### PR DESCRIPTION
**Pull request by BSBandme**
Additional fix for #4299

### CODE REVIEW

File: src/google/protobuf/compiler/js/js_generator.cc [hunk 1]
```diff
@@ -1681,7 +1681,7 @@ void Generator::GenerateProvides(const GeneratorOptions& options,
       if (options.import_style == GeneratorOptions::kImportCommonJsStrict) {
         string namespaceObject = *it;
         // Remove "proto." from the namespace object
-        GOOGLE_CHECK(namespaceObject.compare(0, 6, "proto."));
+        GOOGLE_CHECK(!namespaceObject.compare(0, 6, "proto."));
         namespaceObject.erase(0, 6);
```
Comment (TeBoring) [inline src/google/protobuf/compiler/js/js_generator.cc]: If there is a check for that, do we still need to erase it explicitly?

Comment (BSBandme) [reply src/google/protobuf/compiler/js/js_generator.cc]: Sorry but what do you mean by "erase it explicitly"? Do you mean change to GOOGLE_CHECK for erasing?

File: src/google/protobuf/compiler/js/js_generator.cc [hunk 2]
```diff
@@ -1681,7 +1681,7 @@ void Generator::GenerateProvides(const GeneratorOptions& options,
       if (options.import_style == GeneratorOptions::kImportCommonJsStrict) {
         string namespaceObject = *it;
         // Remove "proto." from the namespace object
-        GOOGLE_CHECK(namespaceObject.compare(0, 6, "proto."));
+        GOOGLE_CHECK(!namespaceObject.compare(0, 6, "proto."));
```
Comment (xfxyjwf) [inline src/google/protobuf/compiler/js/js_generator.cc]: Please do:
```
GOOGLE_CHECK(namespaceObject.compare(0, 6, "proto.") == 0);
```

Treating the return value of compare as bool is apparently confusing and nobody caught the error in the old code.

**Comment by xfxyjwf** [comment]
Thanks for the fix. Can you also help check why it didn't fail on Linux before? Such an error should be caught everywhere.

**Pull request merged**